src/main/java/com/senpro/jafrabackend/services/YelpService.java

```diff
@@ -19,50 +20,74 @@
 @PropertySource("classpath:credentials.properties")
 public class YelpService {
 
-    @Value("${yelp.token}")
-    private String authToken;
-    private String baseUrl;
-    private RestTemplate restTemplate;
-    private HttpHeaders headers;
-
-    @Autowired
-    public YelpService(RestTemplate restTemplate) {
-        this.restTemplate = restTemplate;
-        baseUrl = "https://api.yelp.com/v3";
-    }
-
-    public List<Business> getRestaurants(String type, long latitude, long longitude) throws EntityNotFoundException {
-
-        String url =
-                baseUrl +
-                        "/businesses/search?term=food&latitude=" +
-                        latitude +
-                        "&longitude=" +
-                        longitude +
-                        "&radius=4000&sort_by=distance&limit=10&categories=" +
-                        type;
-
-        setHeaders();
-        ResponseEntity<BusinessResponseWrapper> response = restTemplate.exchange(
-                url,
-                HttpMethod.GET,
-                new HttpEntity<>("parameters", headers),
-                BusinessResponseWrapper.class
-        );
-
-        BusinessResponseWrapper businessWrapper = response.getBody();
-        validateResponse(businessWrapper);
-        return businessWrapper.getBusinesses();
-    }
-
-    private void setHeaders() {
-        headers = new HttpHeaders();
-        headers.add("Authorization", "Bearer " + authToken);
-    }
-
-    private void validateResponse(BusinessResponseWrapper wrapper) throws EntityNotFoundException {
-        if (wrapper == null) throw new EntityNotFoundException("Restaurants");
-        if (wrapper.getBusinesses() == null) throw new EntityNotFoundException("Restaurants");
-        if (wrapper.getBusinesses().size() == 0) throw new EntityNotFoundException("Restaurants");
-    }
+  @Value("${yelp.token}")
+  private String authToken;
+
+  private String baseUrl;
+  private RestTemplate restTemplate;
+  private HttpHeaders headers;
+
+  @Autowired
+  public YelpService(RestTemplate restTemplate) {
+    this.restTemplate = restTemplate;
+    baseUrl = "https://api.yelp.com/v3";
```

Comment:
nit: Could this be moved to come properties file as well instead of being hard coded?